pkg/controller/controller.go

```diff
@@ -549,20 +749,72 @@ func (s *Service) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest
 			}
 		}
 
+		// used to restore session state after removing the volume
+		restoreMetroSession := func() error { return nil }
+
 		if len(vgs.VolumeGroup) != 0 {
 			// Remove volume from volume group
+			// Metro replication needs to be paused to remove the volume from the group.
+			var metroSessionID string
+			if isMetro {
+				isMetroVolumeGroup = true
+				metroSession, err := arr.Client.GetReplicationSessionByID(ctx, vgs.VolumeGroup[0].MetroReplicationSessionID)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "unable to get metro session %s: %s", vgs.VolumeGroup[0].MetroReplicationSessionID, err.Error())
+				}
+
+				metroSessionID = metroSession.ID
```

Comment:
metroSessionID could also be initialized with vgs...MetroSessionID, before line 761.

Reply:
I actually just ran into this issue during testing.
It turns out, `GetVolumeGroupsByVolumeID()` does not return the MetroSessionID. So it wasn't able to even get the replication session because the metro ID was empty.

I'm changing `GetReplicationSessionByID()` to use `GetReplicationSessionByLocalResourceID(..., vg.ID)` at the moment and re-testing.

It would be possible to update the query params for getting volume groups with volume ID, but that will require a change to gopowerstore.